### PR TITLE
Store events in localStorage so they are not lost if the user loses network connectivity

### PR DESCRIPTION
**Pull request by spyro92**

### CODE REVIEW

File: src/browser.js [hunk 1]
```diff
@@ -40,29 +38,38 @@ class UniversalEmitter extends Emitter {
...
   }
   // match server api
   from(): UniversalEmitter {
     return this;
   }
-  flushInternal(): void {
-    if (this.batch.length > 0) {
-      this.fetch('/_events', {
+  async flushInternal(): Promise<void> {
+    const items = getAndClear();
```
Comment (dennisgl) [inline src/browser.js]: Wonder why are we clearing before ensuring the batch is sent. 
`empty -> send -> add back if fail` vs `send -> clear if success` seems like the latter is safer?

File: src/event-storage.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  } catch (e) {
+    // storage may not be writable, do nothing
+  }
+};
+
+export const set = (events: EventType[]) => {
+  try {
+    window.localStorage.setItem(storageKey, JSON.stringify(events));
+  } catch (e) {
+    // storage might be full or not writable, do nothing
```
Comment (dennisgl) [inline src/event-storage.js]: seems like we lose the events if we somehow cannot write the store and prevent them for being sent... Not sure if that's robust?

File: src/event-storage.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+/* global window */
+
+const storageKey = 'fusion-events';
+
+type EventType = {|
+  type: mixed,
+  payload: mixed,
+|};
+
+// localStorage wrappers *exported only for testing*
```
Comment (dennisgl) [inline src/event-storage.js]: Can we make a class for the localStorage interface and ideally typed? btw not sure what does "exported only for testing" mean when we actually need them in the browser plugin

**Comment by dennisgl** [comment]
In general, I think standardizing a client event batch/store is a good idea. Few points to potentially make it better:
- We should define a generic and flexible interface for such event batch/store.
some very rough idea:
```js
class ClientEventBatch {
  pushFront() {}
  pushBack() {} // alias: push()
  getAll() {}
  get(eventType) {}
  drop(eventType) {} //or remove(eventType)
  clear() {}
}
```

The internal store should always fall back to a simple array for robustness.

- default implementation based on `localStorage` is fine, just look out for size limit and other gotchas.

File: src/browser.js (hunk 1 above)
Comment (spyro92) [reply src/browser.js]: If the request takes a really long time, new events could have been added to the list - so its unsafe just to empty the array at the end.

File: src/event-storage.js (hunk 2 above)
Comment (spyro92) [reply src/event-storage.js]: Thats a good point, I'll look into this.

File: src/event-storage.js (hunk 3 above)
Comment (spyro92) [reply src/event-storage.js]: The exports at the bottom are public and the ones at the top are pseudo private...

I like the interface idea. Will explore further tomorrow!

**Pull request closed without merging**